File: frontend/src/types/context.ts
import type { PermissionCode } from './permissions';

export interface OrganizationEnvironment {
  type: 'organization';
  organizationId: string;
  organizationName: string;
}

export interface PlatformEnvironment {
  type: 'platform';
}

export type ActiveEnvironment = OrganizationEnvironment | PlatformEnvironment | null;

export interface OrganizationContext {
  organization_id: string;
  organization_name: string;
  permissions: PermissionCode[];
}

export interface PlatformContext {
  permissions: PermissionCode[];
}

export interface UserContextResponse {
  organizations: OrganizationContext[];
  platform: PlatformContext | null;
}

File: backend/app/modules/core_data/schemas/context.py
"""User context schemas — environments and permissions."""

from uuid import UUID

from pydantic import BaseModel


class OrganizationEnvironment(BaseModel):
    """Organization context: ID, name, and user's permissions in it."""

    organization_id: UUID
    organization_name: str
    permissions: list[str]


class PlatformEnvironment(BaseModel):
    """Platform-level context: user's global (super admin) permissions."""

    permissions: list[str]


class UserContextResponse(BaseModel):
    """Complete user access context: all organizations + platform permissions."""

    organizations: list[OrganizationEnvironment]
    platform: PlatformEnvironment | None
